pom.xml

```diff
@@ -504,7 +504,7 @@
           <creationTime>USE_CURRENT_TIMESTAMP</creationTime>
           <entrypoint>${cryostat.entrypoint}</entrypoint>
           <mainClass>${cryostat.mainClass}</mainClass>
-          <extraClasspath>${env.CLASSPATH}</extraClasspath>
+          <extraClasspath>${env.CLASSPATH}:${env.CRYOSTAT_CLIENTLIB_PATH}</extraClasspath>
```

Comment:
I think this should just be a hardcoded string (defined in a Maven property maybe, or else just hardcoded here). We want/need this to be very repeatable across different build environments, whether that's your laptop or my desktop or the GitHub Actions CI builder, so all of these inputs should be consistent and not come from the builder's environment.

Basically, consider that the environment witnessed by the builder reading the `pom.xml` is certainly a different environment from that witnessed by the Cryostat instance running inside the built image later on.

Reply:
If environment variables that need to be consistent across all builder environments are hardcoded, why are some properties defined in `pom.xml` and others are defined in the shell files?

Reply:
By "the shell files", do you mean `entrypoint.sh`? Or do you mean things like `smoketest.sh` and `run.sh`?

`pom.xml` is only used by Maven for building the product (compiling Java, building web-client, assembling the OCI image) and running integration tests against that product. The various Maven properties defined and used here are all part of the build system, not part of the Cryostat runtime, and AFAIK none of it currently depends on the environment variables present on the host (builder, who runs mvn) machine. Some of those properties and configurations may be used to set environment variables that are passed on to the OCI container during integration testing, for example, but this is a different and later stage than building the image.

`entrypoint.sh` is one of the pieces that gets put inside that OCI image and is what is initially run within the container when it's started, so it's part of the runtime and is completely distinct/independent of the build environment. It checks the runtime environment and does some setup (ex. SSL/TLS cert import) before starting the actual Cryostat Java process.

`smoketest.sh`/`run.sh` set environment variables that are provided to the OCI container at runtime, and those environment variables end up read by `entrypoint.sh` or the Cryostat Java process within the container.

Reply:
To give a more concrete illustration of why I'm against setting `${env.CLASSPATH}` here, this would mean that I can do:

`CLASSPATH=/home/aazores/workspace/cryostat/someExtraClasspath mvn package`

and this would produce a Cryostat image for me where Jib generates a classpath file that contains the string `/home/aazores/workspace/cryostat/someExtraClasspath` as part of the application's runtime classpath. But for anyone else who wants to run this image, that's a meaningless path. Even worse, for *myself* it's a meaningless path, because I set it using an absolute path for my development machine, but this path definitely doesn't existing within the Cryostat image. If I change that to some path that does exist in the image it still doesn't do what it sounds like, because Jib will add that path to the classpath, but Maven/Jib doesn't actually copy the contents of that directory into the image, and so at runtime Cryostat will add that path to the classpath but there will still be nothing in there. Or there might be things in there, but not what I thought there would be at build time, because my build environment is in an independent time/place from the runtime environment Cryostat finds itself in.

If I want to actually include things into the classpath I would not only need to set this `CLASSPATH` env var, I would also need to configure Jib to include the directory into the built image (that's what `src/main/extras` is). But if that's the case, then that path is well-known within the pom.xml already since it has to be there for the Jib plugin to include it, so defining it by env var means either it can/will be overridden by an incorrect value, or it will be empty by default even though the correct value is already known.

Reply:
Thank you, I was wondering why `run.sh` and `smoketest.sh` set environment variables. This makes more sense now.

From the github gist above, it looks like Jib's computed classpath is the same as the classpath defined in `entrypoint.sh` (`/app/resources:/app/classes:/app/libs/*`), which means I don't think I need to include `{env.CLASSPATH}` to `<extraClasspath>`.

Reply:
Makes sense. In the other conversation thread above I also explained that I don't think the `clientLibPath` needs to be added to `extraClasspath` either. Doing it that way requires the path to be determined at build time only, which removes some flexibility for end users who are running/deploying Cryostat. It isn't a hard requirement that that path is configurable, but I think it makes some sense to allow it to be. It could save an end user some time and effort if they want to build their own Cryostat using a different base image which already has a directory containing some client libraries, for example.